lightningd/peer_failed.c

```diff
@@ -30,6 +30,8 @@ void peer_failed(int peer_fd, struct crypto_state *cs,
 	va_start(ap, fmt);
 	errmsg = tal_vfmt(NULL, fmt, ap);
 	va_end(ap);
+	/* Make sure it's correct length for error. */
+	tal_resize(&errmsg, strlen(errmsg)+1);
```

Comment:
I ran into this trap, but *no*, it's not!  valgrind pinged me for sending uninitialzied bytes.  Maybe it should....